docs/madr/decisions/035-mesh-metric-policy.md

```diff
@@ -0,0 +1,126 @@
+      usedOnly: true
+    aggregate:
+      - path: "/metrics/prometheus" # is name still needed since targetRef can just point to a name?
+        port: 8888
+      - port: 8000 # default path is /metrics
+    backends:
+      - type: Prometheus
+        prometheus:
+          port: 5670
+          path: /metrics
+          tls:
+            mode: disabled | providedTLS | activeMTLSBackend
+      - type: OpenTelemetry
+        openTelemetry:
+          endpoint: otel-collector:4317
+```
+
+### Configure MADS based on policy data
+MADS server is responsible for aggregating information about dataplanes needed for Prometheus monitoring. 
+It now extracts information from the `Mesh` resource which proxy should be monitored and creates a `MonitoringAssignment` response.
+
+We need to change the source of our information about monitored proxies from `Mesh` to `MeshMetric` policy. We cannot treat 
+new policy as usual in this context, we should not merge policies, but create different `MonitoringAssignment` for each 
+defined policy, this is needed for scenarios like migrating to a new backend. For example, when we define two `MeshMetric` 
+policies that target the same data plane proxy, we should create two separate `MonitoringAssignment` responses. 
+We can achieve this by extending Prometheus integration, so that it sends backend name with requests to MADS API.
+
+### Pushing metrics from DP to OTEL collector
+In order to push metrics to the OTEL collector, and preserve current Prometheus metrics, we need to start using [OTEL metrics sdk](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric). 
+It allows us to export metrics in multiple formats, especially the ones that we are interested in, Prometheus and OTLP. 
+Thanks to this, we will be able to preserve our old functionality and add the possibility to push metrics to the OTEL collector.
+
+### Metrics hijacker
+To preserve metrics hijacker functionality, we need to implement our own OTEL [Producer](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#Producer). 
+It needs to make HTTP requests to Envoy to gather stats, apply our custom transformations and convert them to OTLP format. 
+Thanks to usage of [PeriodicReader](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric#PeriodicReader) 
+from OTEL SDK, we will be able to make these calls periodically. This reader can use both Prometheus and OTLP exporters 
+to publish metrics to desired backends.
+
+### Using xDS to configure DPP
+In order to pick which backend we will be using for metrics collection, we need to somehow configure our data plane dynamically. 
+The easiest way to do this is to use xDS and create an Envoy listener that will serve configuration for DPP, 
+then we will need to periodically make requests from DPP to it to check if configuration changed. This will enable us
+to not only configure metrics backend, but will make DPP easily configurable from control-plane with policies or any other mechanism.
```

Comment:
We need more detail about this in this MADR. What's the API like, what's to see on it...

Reply:
agreed. But it's a clever idea 👍

Reply:
added more details